### ISSUE TITLE
hyperopt fmin for external model/function

### ISSUE THREAD

**Issue by jiacheng-yao** [hyperopt/hyperopt#527]
Hi, all the codes I have seen so far for hyperopt require that fmin is defined and well described in the code it self, such as an explicit function or a keras/tensorflow model. But what about the case when the function I want to optimize is the result of a complex external program from another programming language that gets back the loss value after many hours? I mean is it possible to make hyperopt choose e.g. 10 parameter sets randomly and let the external program run 10 times with the 10 parameter sets and give back the loss values for the 10 runs, and then let hyperopt output the next sample with the highest expected improvement(EI)?

**Comment by dandxy89** [comment]
You could consider writing a little helper function with calls a bash script or bat file (Windows) from Python, if you wanted the whole thing to be automated. The script would then pass the parameters from Python to the script which would then invoke the 'external program'. Assuming you're able to get the 'external program' to display its output via stdout you could then easily read it.

```
from subprocess import Popen
p = Popen("batch.bat", cwd=r"C:\Path\to\batchfolder")
stdout, stderr = p.communicate()
```

Just a suggestion...
